RadixWallet/Features/AssetTransferFeature/Components/ChooseAccount/ChooseReceivingAccount+View.swift

```diff
@@ -1,141 +1,113 @@
 		@FocusState private var focusedField: Bool
 
 		public init(store: StoreOf<ChooseReceivingAccount>) {
 			self.store = store
 		}
-	}
-}
 
-extension ChooseReceivingAccount.View {
-	public var body: some View {
-		NavigationStack {
-			WithViewStore(store, observe: ChooseReceivingAccount.ViewState.init(state:), send: { .view($0) }) { viewStore in
-				ScrollView {
-					VStack(spacing: .medium2) {
-						Text(L10n.AssetTransfer.ChooseReceivingAccount.enterManually)
-							.textStyle(.body1Regular)
-							.foregroundColor(.app.gray1)
+		public var body: some SwiftUI.View {
+			WithPerceptionTracking {
+				NavigationStack {
+					ScrollView {
+						VStack(spacing: .medium2) {
+							Text(L10n.AssetTransfer.ChooseReceivingAccount.enterManually)
+								.textStyle(.body1Regular)
+								.foregroundColor(.app.gray1)
 
-						addressField(viewStore)
+							addressField
 
-						Divider()
+							if !store.chooseAccounts.availableAccounts.isEmpty {
+								Divider()
 
-						Text(L10n.AssetTransfer.ChooseReceivingAccount.chooseOwnAccount)
+								Text(L10n.AssetTransfer.ChooseReceivingAccount.chooseOwnAccount)
+							}
 
-						ChooseAccounts.View(
-							store: store.scope(
-								state: \.chooseAccounts,
-								action: { .child(.chooseAccounts($0)) }
+							ChooseAccounts.View(
+								store: store.scope(state: \.chooseAccounts, action: \.child.chooseAccounts)
 							)
-						)
-						.opacity(viewStore.canSelectOwnAccount ? 1.0 : 0.6)
-						.disabled(!viewStore.canSelectOwnAccount)
-					}
-					.padding(.medium3)
-				}
-				.destinations(with: store)
-				.footer { chooseButton(viewStore) }
-				.radixToolbar(title: L10n.AssetTransfer.ChooseReceivingAccount.navigationTitle)
-				.toolbar {
-					ToolbarItem(placement: .navigationBarLeading) {
-						CloseButton {
-							store.send(.view(.closeButtonTapped))
+							.opacity(store.canSelectOwnAccount ? 1.0 : 0.6)
+							.disabled(!store.canSelectOwnAccount)
 						}
+						.padding(.medium3)
+					}
+					.destinations(with: store)
+					.footer { chooseButton }
+					.radixToolbar(title: L10n.AssetTransfer.ChooseReceivingAccount.navigationTitle) {
+						store.send(.view(.closeButtonTapped))
 					}
 				}
 			}
 		}
-	}
 
-	private func addressField(_ viewStore: ViewStoreOf<ChooseReceivingAccount>) -> some View {
-		AppTextField(
-			placeholder: L10n.AssetTransfer.ChooseReceivingAccount.addressFieldPlaceholder,
-			text: viewStore.binding(
-				get: \.manualAccountAddress,
-				send: { .manualAccountAddressChanged($0) }
-			),
-			hint: viewStore.manualAddressHint,
-			focus: .on(
-				true,
-				binding: viewStore.binding(
-					get: \.manualAccountAddressFocused,
-					send: { .focusChanged($0) }
+		private var addressField: some SwiftUI.View {
+			AppTextField(
+				placeholder: L10n.AssetTransfer.ChooseReceivingAccount.addressFieldPlaceholder,
+				text: $store.manualAccountAddress.sending(\.view.manualAccountAddressChanged),
```

Comment:
Do you have to manually specify the `sending` action even though you are using Bindable?

Reply:
It seems so. The `sending` function converts the value to a `Binding`.

Reply:
Alright, it seems that with observation you would still need to add the BindingReducer etc if you want the plain binding without `send`:
```
@Reducer
struct Settings {
  @ObservableState
  struct State { /* ... */ }
  enum Action: BindableAction { 
    case binding(BindingAction<State>)
   /* ... */ 
  }

  var body: some Reducer<State, Action> {
    BindingReducer()
  }
}
```
Not worth it for a single binding, and we would need to rearchitect the FeatureReducer for it to work.